clickhouse_search/search_tests.py

```diff
@@ -0,0 +1,1150 @@
+import os
+
+from clickhouse_search.test_utils import VARIANT1, VARIANT2, VARIANT3, VARIANT4, CACHED_VARIANTS_BY_KEY
+from seqr.models import Project
+from seqr.utils.search.search_utils_tests import SearchTestHelper
+from seqr.utils.search.utils import query_variants
+from seqr.views.utils.json_utils import DjangoJSONEncoderWithSets
+
+
+@mock.patch('clickhouse_search.search.CLICKHOUSE_SERVICE_HOSTNAME', 'localhost')
+class ClickhouseSearchTests(SearchTestHelper, TestCase):
+    databases = '__all__'
+    fixtures = ['users', '1kg_project', 'reference_data', 'clickhouse_search', 'clickhouse_transcripts']
+
+    @classmethod
+    def setUpTestData(cls):
+        with connections['clickhouse'].cursor() as cursor:
+            cursor.execute("""
+            CREATE DICTIONARY IF NOT EXISTS "GRCh38/SNV_INDEL/gt_stats_dict"
+            (
+                key UInt32,
+                ac UInt32,
+                an UInt32,
+                hom UInt32,
+            )
+            PRIMARY KEY key
+            SOURCE(CLICKHOUSE(
+                USER %s
+                PASSWORD %s
+                QUERY "SELECT * FROM VALUES ((1, 9, 90, 2), (2, 28, 90, 4), (3, 4, 6, 1), (4, 2, 90, 0))"
+            ))
+            LIFETIME(0)
+            LAYOUT(FLAT(MAX_ARRAY_SIZE 500000000))
+            """, [os.environ.get('CLICKHOUSE_USER', 'clickhouse'), os.environ.get('CLICKHOUSE_PASSWORD', 'clickhouse_test')])
+
+    def setUp(self):
+        super().set_up()
+        Project.objects.update(genome_version='38')
+
+    def _assert_expected_search(self, expected_results, gene_counts=None, **search_kwargs):
+        self.search_model.search.update(search_kwargs or {})
+
+        variants, total = query_variants(self.results_model, user=self.user)
+        encoded_variants = json.loads(json.dumps(variants, cls=DjangoJSONEncoderWithSets))
+
+        self.assertListEqual(encoded_variants, expected_results)
+        self.assertEqual(total, len(expected_results))
+        self._assert_expected_search_cache(encoded_variants, total)
+
+    def _assert_expected_search_cache(self, variants, total):
+        cached_variants = [CACHED_VARIANTS_BY_KEY[variant['key']] for variant in variants]
+        results_cache = {'all_results': cached_variants, 'total_results': total}
+        self.assert_cached_results(results_cache)
+
+    def test_single_family_search(self):
+        with self.assertRaises(NotImplementedError):
+            query_variants(self.results_model, user=self.user)
+
+        self.results_model.families.set(self.families.filter(guid='F000002_2'))
+        variant_gene_counts = {
+            'ENSG00000097046': {'total': 2, 'families': {'F000002_2': 2}},
+            'ENSG00000177000': {'total': 2, 'families': {'F000002_2': 2}},
+            'ENSG00000277258': {'total': 1, 'families': {'F000002_2': 1}}
+        }
+        self._assert_expected_search(
+            [VARIANT1, VARIANT2, VARIANT3, VARIANT4], gene_counts=variant_gene_counts,
+        )
+
```

Comment:
everything from here down is copied from the hail search tests with the small modification to change it from async to sync syntax. The plan is to uncomment tests as development progresses